### ISSUE TITLE
Persistent docker doesn't work with AgentSkills in test 

### ISSUE THREAD

**Issue by li-boxuan** [All-Hands-AI/OpenHands#2176]
@SmartManoj could you please take a look at this?

<img width="963" alt="image" src="https://github.com/OpenDevin/OpenDevin/assets/25746010/ef402432-2947-4f50-bdc3-3a8b6b26134d">

Persistent docker thing doesn't seem to work well with agent skills. Repro command:

```bash
ONLY_TEST_NAME=test_edits TEST_ONLY=true ONLY_TEST_AGENT=CodeActAgent ./tests/integration/regenerate.sh
```

(with `persist_sandbox = true`)

I think https://github.com/OpenDevin/OpenDevin/pull/1998 causes a regression.

The test passed because you set `persist_sandbox = false` in CI.

If we make this persistence thing by default ON, we shall probably also turn it on in CI?

_Originally posted by @li-boxuan in https://github.com/OpenDevin/OpenDevin/issues/2139#issuecomment-2138723188_

**Comment by li-boxuan** [comment]
Creating an issue to track the problem

**Comment by SmartManoj** [comment]
works if `workdir='/'`

![image](https://github.com/OpenDevin/OpenDevin/assets/7231077/920e594d-a4f1-490a-bcbc-84fca6161ac7)

---
[Action link](https://github.com/SmartManoj/Kevin/actions/runs/9312939623/job/25634527255)

`ls -la /workspace` returns just `total 0\n`

The typical two lines are missing. Open to suggestions!

![image](https://github.com/OpenDevin/OpenDevin/assets/7231077/25eda074-e292-43a9-aa10-7ec67fb14c27)

---

**Comment by SmartManoj** [comment]
> works if `workdir='/'`.

[New issue ](https://github.com/OpenDevin/OpenDevin/pull/2016/files#r1623799719)pip not found raised after this.

Are there any tests that clear sandbox storage?

**Comment by SmartManoj** [comment]
Has anyone experienced this behavior after running the test?

![image_2024-06-05_11-03-40](https://github.com/OpenDevin/OpenDevin/assets/7231077/2d27785b-725a-49de-87a7-b6028318d6a8)

**Comment by tobitege** [comment]
> Has anyone experienced this behavior after running the test?

`regenerate.sh` removes for pretty much every test the `_test_workspace` folder before starting each single test (rm -rf).
If a test fails, it also does that. And even if you're in "OpenDevin" in terminal, after the test it thinks it is in the now removed test folder. If you manually go a folder back and forward, it'll reset. 
If not done that way, you get that "FileNotFoundError" running regenerate again.
That's at least my experience here the last days and it _looks_ pretty close to what happens in your screenshot?

**Comment by mamoodi** [comment]
@li-boxuan can this issue be closed now that PERSIST_SANDBOX is set to false by default and likely needs an overhaul if implemented correctly?

**Comment by li-boxuan** [comment]
@mamoodi I think we should keep this issue open unless we fix it, remove PERSIST_SANDBOX completely, OR have another issue/PR that tracks an overhaul?

**Comment by mamoodi** [comment]
Makes sense. Let's keep it open until one of those happen. I think what needs to be done is a little bit in the air.
